### DISCUSSION TITLE
HEAT and Guid Generation

### DISCUSSION THREAD

**Discussion by Steve0212a** [wixtoolset/issues#7378] [Questions] [answered]
I am converting an installer from v3 to v4 that uses heat and is not installing in a standard folder.  This led to a lot of these errors:

`error WIX0231: The component 'cmpXoPJCv6AJKpzfum_tcEPpdYGCNU' has a key file with path 'TARGETDIR\.\xxx-s\system.composition.hosting.dll'. Since this path is not rooted in one of the standard directories (like ProgramFilesFolder), this component does not fit the criteria for having an automatically generated guid. (This error may also occur if a path contains a likely standard directory such as nesting a directory with name "Common Files" under ProgramFilesFolder.)
`

I have tried the following two options to fix it, but Heat is still generating the files with Guid="*" as shown here
```
            <Component Id="cmpXoPJCv6AJKpzfum_tcEPpdYGCNU" Guid="*">
                <File Id="filWNF6Di_uJs0w0W6w.kE0orwVOEE" KeyPath="yes" Source="$(SourceFolder)\System.Composition.Hosting.dll" />
            </Component>
```

**Option 1: define constants**
`<DefineConstants>HarvestAutogenerateGuids=false;HarvestGenerateGuidsNow=true</DefineConstants>`

**Option 2:  specify in HarvestDirectory node**
```
    <HarvestDirectory Include="xxx">
      <ComponentGroupName>ContainerComponentGroup</ComponentGroupName>
      <DirectoryRefId>Containers</DirectoryRefId>
      <PreprocessorVariable>SourceContainerFolder</PreprocessorVariable>
      <SuppressRootDirectory>true</SuppressRootDirectory>
      <SuppressFragments>true</SuppressFragments>
      <HarvestDirectorySuppressAllWarnings>true</HarvestDirectorySuppressAllWarnings>
      <HarvestDirectorySuppressCom>true</HarvestDirectorySuppressCom>
      <HarvestDirectorySuppressRegistry>true</HarvestDirectorySuppressRegistry>
      <HarvestDirectoryAutogenerateGuids>false</HarvestDirectoryAutogenerateGuids>
      <HarvestDirectoryGenerateGuidsNow>true</HarvestDirectoryGenerateGuidsNow>
    </HarvestDirectory>
```

What am I doing wrong to get Heat to generate specific Guids instead of "*"?

thx

**Comment by robmen** [comment, marked as answer]
Don't. See [Directory ComponentGenerationSeed attribute](https://wixtoolset.org/docs/schema/wxs/directory/#attributes).

**Comment by Steve0212a** [comment]
that solved it - thanks for the tip.  May want to add that note into this doc page --> https://wixtoolset.org/docs/tools/heat/

**Reply by robmen** [reply]
There is a link at the bottom of every doc page to help improve it. If you think you just learned something useful to others, try to capture it.

Personally, I don't care about heat.
